src/GridSquare.svelte

```diff
@@ -5,12 +5,31 @@
   export let isActive: boolean;
   export let isPartOfCurrentClue: boolean;
   export let square: Square;
+  export let updateCell: (newValue: string) => void;
 
   let inputElement: HTMLInputElement | null;
   $: if (isActive && inputElement) {
     inputElement.focus();
   }
 
+  /**
+   * Handles the input event for the crossword input.
+   * @param {Event & { currentTarget: EventTarget & HTMLInputElement }} event - The input event.
+   */
+  function handleInput(
+    event: Event & { currentTarget: EventTarget & HTMLInputElement },
+  ) {
+    const target = event.currentTarget;
+    if (target) {
+      // todo check delete works
+      const inputChar = target.value
+        ? target.value.slice(-1).toUpperCase()
```

Comment:
Hmm maybe you could also put something on the `input` element to ensure it only ever has length one?

https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/maxlength